### PR TITLE
8342958: Use jvmArgs consistently in microbenchmarks

### PR DESCRIPTION
**Pull request by cl4es**
Many OpenJDK micros use `@Fork(jvmArgs/-Append/-Prepend)` to add JVM reasonable or necessary flags, but when deploying and running micros we often want to add or replace flags to tune to the machine, test different GCs, etc. The inconsistent use of the different `jvmArgs` options make it error prone, and we've had a few recent cases where we've not been testing with the expected set of flags. 

This PR suggests using `jvmArgs` consistently. I think this aligns with the intuition that when you use `jvmArgsAppend/-Prepend` intent is to add to a set of existing flags, while if you supply `jvmArgs` intent is "run with these and nothing else". Perhaps there are other opinions/preferences, and I don't feel strongly about which to consolidate to as long as we do so consistently. One argument could be made to consolidate on `jvmArgsAppend` since that one is (likely accidentally) the current most popular (143 compared to 59 `jvmArgsPrepend` and 18 `jvmArgs`).

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8342958](https://bugs.openjdk.org/browse/JDK-8342958): Use jvmArgs consistently in microbenchmarks (**Enhancement** - P4)

### Reviewers
 * [Eric Caspole](https://openjdk.org/census#ecaspole) (@ericcaspole - Committer)
 * [Jorn Vernee](https://openjdk.org/census#jvernee) (@JornVernee - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/21683/head:pull/21683` \
`$ git checkout pull/21683`

Update a local copy of the PR: \
`$ git checkout pull/21683` \
`$ git pull https://git.openjdk.org/jdk.git pull/21683/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 21683`

View PR using the GUI difftool: \
`$ git pr show -t 21683`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/21683.diff">https://git.openjdk.org/jdk/pull/21683.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/21683#issuecomment-2435372202)

### CODE REVIEW

**Review by ericcaspole — APPROVED** [review]
Looks good.

**Comment by cl4es** [comment]
Yes, that's typically what we do in testing, the subtle problem solved by this is that when you do so you'll silently overwrite pre-existing jvmArgsAppend annotation. I think the suggested scheme is more intuitive in that regard.

**Review by JornVernee — APPROVED** [review]
Overall I like the idea. I took a look at the `foreign` benchmarks in particular, and don't see any issues with them.

File: test/micro/org/openjdk/bench/vm/compiler/overhead/SimpleRepeatCompilation.java [hunk 1]
```diff
@@ -95,25 +95,25 @@ public int mixHashCode(String value) {
      = "-XX:CompileCommand=option,org/openjdk/bench/vm/compiler/overhead/SimpleRepeatCompilation.trivialMath,intx,RepeatCompilation,2000";
 
     @Benchmark
-    @Fork(jvmArgsAppend={"-Xbatch",TRIVIAL_MATH_METHOD})
+    @Fork(jvmArgs={"-Xbatch",TRIVIAL_MATH_METHOD})
```
Comment (turbanoff) [inline test/micro/org/openjdk/bench/vm/compiler/overhead/SimpleRepeatCompilation.java:98]:
```suggestion
    @Fork(jvmArgs={"-Xbatch", TRIVIAL_MATH_METHOD})
```

File: test/micro/org/openjdk/bench/vm/compiler/overhead/SimpleRepeatCompilation.java [hunk 2]
```diff
@@ -135,25 +135,25 @@ public int trivialMath(int a, int b) {
      = "-XX:CompileCommand=option,org/openjdk/bench/vm/compiler/overhead/SimpleRepeatCompilation.largeMethod,intx,RepeatCompilation,100";
 
     @Benchmark
-    @Fork(jvmArgsAppend={"-Xbatch",LARGE_METHOD})
+    @Fork(jvmArgs={"-Xbatch",LARGE_METHOD})
```
Comment (turbanoff) [inline test/micro/org/openjdk/bench/vm/compiler/overhead/SimpleRepeatCompilation.java:138]:
```suggestion
    @Fork(jvmArgs={"-Xbatch", LARGE_METHOD})
```

Comment (cl4es) [reply test/micro/org/openjdk/bench/vm/compiler/overhead/SimpleRepeatCompilation.java:138]: I don't think this PR is the place to address pre-existing and non-consequential style issues.

**Comment by eme64** [comment]
Hi Claes,
I'm having some strange results with your JMH PR:
https://github.com/openjdk/jdk/pull/21683
And my benchmark:
https://github.com/openjdk/jdk/blame/master/test/micro/org/openjdk/bench/vm/compiler/VectorLoadToStoreForwarding.java
Running it like this:
`make test TEST="micro:vm.compiler.VectorLoadToStoreForwarding.*benchmark_03" CONF=linux-x64`
And I'm getting this result:
```
Benchmark                                                                        (SIZE)  (seed)  Mode  Cnt     Score    Error  Units
VectorLoadToStoreForwarding.VectorLoadToStoreForwardingNoSuperWord.benchmark_03    2048       0  avgt   10  4563.110 ± 64.988  ns/op
VectorLoadToStoreForwarding.VectorLoadToStoreForwardingSuperWord.benchmark_03      2048       0  avgt   10  4549.337 ± 32.239  ns/op
```
But when I revert your change, back to `jvmArgsPrepend` :
```
Benchmark                                                                        (SIZE)  (seed)  Mode  Cnt     Score    Error  Units
VectorLoadToStoreForwarding.VectorLoadToStoreForwardingNoSuperWord.benchmark_03    2048       0  avgt   10  1040.538 ± 14.660  ns/op
VectorLoadToStoreForwarding.VectorLoadToStoreForwardingSuperWord.benchmark_03      2048       0  avgt   10  4533.227 ±  5.161  ns/op
```
It seems that with your PR, the flags don't have the intended effect any more!


With your change, I see this:
`# VM options: -Djava.library.path=/oracle-work/jdk-fork6/build/linux-x64/images/test/micro/native`
And reverting back, I see this:
`# VM options: -XX:-UseSuperWord -Djava.library.path=/oracle-work/jdk-fork6/build/linux-x64/images/test/micro/native`

**Comment by cl4es** [comment]
An oversight, but I forgot to change the `RunTests.gmk`. Filed https://bugs.openjdk.org/browse/JDK-8343345

**Pull request closed without merging**